src/main/java/Deadline.java

```diff
@@ -1,2 +1,13 @@
-package PACKAGE_NAME;public class Deadline {
+public class Deadline extends Task {
+    protected String by;
```

Comment:
Minor nitpick with regards to code readability, it's ok if not changed since it is within a class.
Instead of naming your stored String as "by", could name it something more descriptive like deadlineDate or date since when reviewers when Ctrl + F looking the date stored in your class, they might not think to search for "by" since it is a common word used in many classes, functions and comments,